Add tests for entry type method generation

snoble opened the pull request:
This PR adds tests to verify that the generated SDK exposes type-specific methods (e.g., PostUserFundsAccount, PostFundingSettlement) for entry types defined in schemas with lines.

## Changes
- Added tests for entry type methods with different naming conventions:
  - Hyphenated naming (user-funds-account → PostUserFundsAccount)
  - CamelCase naming (fundingSettlement → PostFundingSettlement)
  - Underscore naming (payment_processing → PostPaymentProcessing)
- Tests verify method existence and functionality when methods are generated
- Tests use schemas with entry types that have lines defined (not empty)
- Consolidated tests into basic-test.test.ts to match existing test structure

## Testing
Tests will verify:
- Whether type-specific Post methods exist on the client
- If methods exist, they can be called and create entries correctly
- Multiple entry types in a single schema work correctly

snoble committed:
Add tests for entry type method generation

- Add test file to verify generated Post methods for entry types
- Test hyphenated, camelCase, and underscore naming conventions
- Test schemas with entry types that have lines defined
- Verify method existence and functionality when methods are generated

snoble committed:
Move entry type method tests to basic-test.test.ts

- Consolidate tests into basic-test.test.ts to match existing test structure
- Remove console.log statements to match existing test style
- Simplify test names and variable names to match existing patterns
- Test entry type methods with hyphenated, camelCase, and underscore naming

snoble committed:
Update generated files

snoble committed:
Add documentation for Post<entry> type-specific methods

- Document how type-specific methods are generated for entry types with lines
- Show examples with different naming conventions (hyphenated, camelCase, underscore)
- Explain when methods are available and fallback to addLedgerEntry

snoble committed:
Update tests to handle both parameter styles for Post methods

- Try individual parameters first (amount as direct arg)
- Fall back to parameters object if individual params fail
- All entry types have required 'amount' parameter from {{amount}} template

snoble committed:
Add parameter style detection and verification tests

- Add detectParameterStyle helper that throws if method signature is unexpected
- Tests now explicitly fail if parameter style can't be determined
- Add dedicated test to verify parameter style and ensure wrong style fails
- Tests will catch API changes or incorrect method signatures

snoble committed:
Make tests fail if using wrong parameter signature

- Add EXPECTED_PARAMETER_STYLE constant to document expected signature
- Replace detectParameterStyle with verifyParameterStyle that fails if signature doesn't match
- Tests now explicitly use expected style and fail if it's wrong
- Remove fallback logic - tests will fail if signature is unexpected

snoble committed:
Add test schema with generated SDK and CI verification

- Add tests/fixtures/test-schema.json with entry types
- Generate GraphQL queries using Fragment CLI
- Generate SDK from GraphQL queries
- Add scripts/update-test-schema.sh for easy updates
- Add CI check to verify generated files match schema
- Add tests/generated-sdk.test.ts that imports generated SDK directly
- Update EXPECTED_PARAMETER_STYLE to 'individual' (Fragment CLI generates individual params)
- Update README to show correct parameter style

snoble committed:
Install Fragment CLI via Homebrew in CI

- Install Homebrew on Ubuntu runner
- Install Fragment CLI from fragment-dev/tap
- Use full path to fragment command in verification step

snoble committed:
Use full path to fragment command in verification step

snoble committed:
Add caching for Homebrew and Fragment CLI

- Cache Homebrew installation directory and cache
- Only install Homebrew if not cached
- Check if Fragment CLI is installed before installing
- Set up brew shellenv properly for cached installations

snoble committed:
Remove sentence about methods availability from README

vigneshwerv commented on README.md:
```diff
@@ -81,6 +81,25 @@ await fragment.addLedgerEntryRuntime({
 
+### Post a Ledger Entry using type-specific methods
+
+When your schema defines entry types with lines, the SDK generates type-specific methods for posting those entries. These methods follow the naming convention `Post<EntryType>` where the entry type name is converted to PascalCase.
```
```suggestion
When your Schema defines Ledger Entry types with lines, the SDK generates type-specific methods for posting those entries. These methods follow the naming convention `Post<EntryType>` where the entry type name is converted to PascalCase.
```

snoble committed:
Add typeVersion support to test schema and tests

- Update test schema to include user-funds-account with typeVersion 1 and 2
- Regenerate GraphQL queries and SDK (includes PostUserFundsAccount_v2)
- Add typeVersion to GraphQL query response fields
- Update test to verify both version 1 and version 2 entries can be posted
- Test verifies typeVersion in response matches expected version

snoble committed:
Add fee line to typeVersion 2 with feeAmount parameter

- Add expense-root account to schema
- Add fee-line to typeVersion 2 entry type using {{feeAmount}} parameter
- Adjust asset-line to account for fee: {{amount}} - {{feeAmount}}
- Update test to include expense account and pass feeAmount parameter
- Generated SDK now includes feeAmount as a required parameter for PostUserFundsAccount_v2

snoble committed:
Add documentation for typeVersion in Post methods

- Explain how typeVersion affects method naming
- Show examples of version 1 vs version 2 method calls
- Document that different versions can have different parameters

snoble committed:
Fix: Add typeVersion to GraphQL query response fields

- Add typeVersion field to all entry response blocks in GraphQL queries
- Regenerate SDK to include typeVersion in TypeScript types
- Fixes TypeScript errors in tests that check typeVersion

snoble committed:
Update generation script to add typeVersion to response fields

- Fragment CLI doesn't include typeVersion in response by default
- Add sed command to insert typeVersion after type field in all entry blocks
- This ensures generated SDK includes typeVersion in TypeScript types
- Regenerate files to match CI expectations

snoble committed:
Remove typeVersion from tests - verify versions by line structure instead

- Remove sed command that added typeVersion to GraphQL queries
- Tests now verify version differences by checking line structure
- Version 1 has 2 lines, version 2 has 3 lines (including fee-line)
- This tests what Fragment CLI actually generates, not modified output

snoble committed:
Update README.md

Co-authored-by: Vigneshwer Vaidyanathan <vignesh.vaid@gmail.com>
